### ISSUE TITLE
Support credential helpers in docker/jib

### ISSUE THREAD

**Issue by SevereOverfl0w** [juxt/pack.alpha#52]
https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#authentication-methods

https://static.javadoc.io/com.google.cloud.tools/jib-core/0.10.0/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.html

Do we also need to have some way to specify whether to use the helper or the docker config file? Should we also support overriding the location for the docker config file?

**Comment by viesti** [comment]
Good questions :) I don't remember if I had deep really deep thoughts on registry authentication. Seems that [Jib FAQ](https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#what-should-i-do-when-the-registry-responds-with-unauthorized) doesn't recommend passing specific credentials ("Specific credentials (not recommend)" in the faq). Then there is an interesting function [inferCredentialHelper()](https://static.javadoc.io/com.google.cloud.tools/jib-core/0.10.0/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.html#inferCredentialHelper--) also :)

**Comment by SevereOverfl0w** [comment]
Yeah, I am a little hesitant about the existence of that, but I guess it's easier to hook into other tools (e.g. `gpg -d ~/foo.gpg`) so it's not terrible. We should probably figure out how to emphasize the other mechanisms first.
